### PR TITLE
Cleaning up lib/date

### PR DESCRIPTION
**Pull request by viiru-**
This converts the utility functions hidden as methods of Date into actual functions, and cleans up the imports. The callers are not converted, but if you prefer I can add that.

### CODE REVIEW

**Comment by viiru-** [comment]
The latest push removes all instances of the Date().<utility function> pattern from the code. There are still cases that use the methods, but in those cases a date object already exists.

**Comment by viiru-** [comment]
In my understanding this branch is now complete and ready to merge. I look forward to any comments you might have.

File: pytrainer/lib/date.py [hunk 1]
```diff
@@ -22,34 +22,71 @@
...
+def time2string(date):
+    return "%0.4d-%0.2d-%0.2d" %(int(date[0]),int(date[1]),int(date[2]))
+
+def getNameMonth(date):
+    day, daysInMonth = calendar.monthrange(date.year, date.month)
+    monthName = calendar.month_name[date.month]
+    return monthName, daysInMonth
+
+def unixtime2date(unixtime):
+    print unixtime
```
Comment (dgranda) [inline pytrainer/lib/date.py]: Would not be a good ides to get rid of *print* statement in *unixtime2date*?

**Comment by dgranda** [comment]
It also would be a great idea to create some unit testing for Date library once functions are defined there, what do you think?

File: pytrainer/lib/date.py (hunk 1 above)
Comment (viiru-) [inline pytrainer/lib/date.py]: David García Granda <notifications@github.com> writes:
>> +    return int(sec)+(int(min)*60)+(int(hour)*3600)
>> +
>> +def getLocalTZ():
>> +    ''' Returns string representation of local timezone'''
>> +    return datetime.datetime.now(tzlocal()).tzname()
>> +
>> +def time2string(date):
>> +    return "%0.4d-%0.2d-%0.2d" %(int(date[0]),int(date[1]),int(date[2]))
>> +
>> +def getNameMonth(date):
>> +    day, daysInMonth = calendar.monthrange(date.year, date.month)
>> +    monthName = calendar.month_name[date.month]
>> +    return monthName, daysInMonth
>> +
>> +def unixtime2date(unixtime):
>> +    print unixtime
>
> Would not be a good ides to get rid of *print* statement in
> *unixtime2date*?

Sure, why not. I have pushed the change.

So far in this series I had not modified any of the code in lib/date.py,
only reorganized it. There are also commented out debug prints in
getDateTime, should those also be removed or perhaps converted to
logging.debug?

-- 
Arto Jantunen

**Comment by dgranda** [comment]
> So far in this series I had not modified any of the code in lib/date.py, only reorganized it. There are also commented out debug prints in getDateTime, should those also be removed or perhaps converted to logging.debug?

Ooops! I have just seen them.

I would remove all *print* statements: some are already commented out and others are duplicated of *logging.debug*

PS: I moved conversation from [date.py line specific](https://github.com/pytrainer/pytrainer/pull/52#discussion_r45576839) to pull's general one for better readability

**Comment by viiru-** [comment]
David García Granda <notifications@github.com> writes:
>> So far in this series I had not modified any of the code in lib/date.py,
> only reorganized it. There are also commented out debug prints in getDateTime,
> should those also be removed or perhaps converted to logging.debug?
>
> Ooops! I have just seen them.
>
> I would remove all *print* statements: some are already commented out and others are duplicated of *logging.debug*

Ok, I have pushed commits removing the rest of them.

-- 
Arto Jantunen

**Comment by viiru-** [comment]
David García Granda <notifications@github.com> writes:

> It also would be a great idea to create some unit testing for Date library
> once functions are defined there, what do you think?

I pushed a commit adding simple test cases for most of those utility
functions, let me know what you think.

-- 
Arto Jantunen

**Pull request merged**